tardis/model/base.py

```diff
@@ -23,6 +23,113 @@
 logger = logging.getLogger(__name__)
 
 
+class ModelState:
+    def __init__(
+        self,
+        velocity,
+        time_explosion,
+        v_boundary_inner,
+        v_boundary_outer,
+    ):
+        self._v_boundary_inner = None
+        self._v_boundary_outer = None
+        self._velocity = None
+        self.raw_velocity = velocity
+        self.time_explosion = time_explosion
+
+    @property
+    def v_boundary_inner(self):
+        if self._v_boundary_inner is None:
+            return self.raw_velocity[0]
+        if self._v_boundary_inner < 0 * u.km / u.s:
+            return self.raw_velocity[0]
+        return self._v_boundary_inner
+
+    @property
+    def v_boundary_outer(self):
+        if self._v_boundary_outer is None:
+            return self.raw_velocity[-1]
+        if self._v_boundary_outer < 0 * u.km / u.s:
+            return self.raw_velocity[-1]
+        return self._v_boundary_outer
+
+    @property
```

Comment:
no - these don't move over. As discussed only v_inner, v_outer, r_inner, r_outer. Just stick to the simple things and ask.